### ISSUE TITLE
CodeGen; "Useless" Conditionals

### ISSUE THREAD

**Issue by Mornix** [JacquesCarette/Drasil#1184]
Current codegen for GlassBR generates the following conditional block in `DerivedValues.py`:

```python
    if (1) :
        inParams.GTF = inParams.g == "AN"
    elif (4) :
        inParams.GTF = inParams.g == "FT"
    elif (2) :
        inParams.GTF = inParams.g == "HS"
```

While this specific pasted example is in Python, this issue is not only in the Python generated code, but in all examples. Depending on the language the following conditionals will either:
1. Fail type checking 
2. Always take the first branch

While I have not triaged this issue in detail, my guess is codegen is reducing something too much (or incorrectly) resulting in these sort of branches.

**Comment by bmaclach** [comment]
This is also fixed on my finally tagless GOOL fork.

**Comment by Mornix** [comment]
Fixed with #1188 and non-issue in FT GOOL. Closing.
